### PR TITLE
Fix a bug amqp uri using Client.fromSharedAccessSignature

### PR DESCRIPTION
**Pull request by kajiiiro**
I fix a bug in Client.fromSharedAccessSignature(iothub device sdk).
and typo console.error.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@kajiiiro__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by pierreca** [comment]
Hi @kajiiiro, thanks for the PR. good catches :)

per our [contributing guidelines](https://github.com/Azure/azure-iot-sdks/blob/master/contribute.md) we only accept code PRs against the `develop` branch though, not directly on `master`. can you resubmit it? 
I'll make sure to get it through the gate and in as soon as you sign the CLA.

Thanks!

File: node/device/core/lib/client.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ Client.fromSharedAccessSignature = function (sharedAccessSignature, Transport) {
   if (!sharedAccessSignature) throw new ReferenceError('sharedAccessSignature is \'' + sharedAccessSignature + '\'');
 
   var sas = SharedAccessSignature.parse(sharedAccessSignature);
-  var uriSegments = sas.sr.split('/')[0];
+  var uriSegments = sas.sr.split('/');
```
Comment (pierreca) [inline node/device/core/lib/client.js]: Looks like we should add a test in `node/device/core/test/_client.test.js` corresponding to this bug, if you have time feel free to do it, I'll be happy to do it if myself you don't feel like it though.

File: node/device/samples/simple_sample_device_with_sas.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var Amqp = require('azure-iot-device-amqp').Amqp;
```
Comment (pierreca) [inline node/device/samples/simple_sample_device_with_sas.js]: one thing we have been thinking about when refactoring the other sample is rename all transport vars to `Protocol` and then use `Protocol` in the call on line 19. this limits the number of changes a user has to make to test various transports. again, I'll be happy to do it if you'd rather not spend the time Thanks for the new sample, it's very much appreciated!

**Comment by kajiiiro** [comment]
> per our contributing guidelines we only accept code PRs against the develop branch though, not directly on master. can you resubmit it?

Is it mean that I create new PR?
Sorry, I don't know how to resubmit it.


> Looks like we should add a test in node/device/core/test/_client.test.js corresponding to this bug, if you have time feel free to do it, I'll be happy to do it if myself you don't feel like it though.
> Add a line note

I try to add it.

**Comment by pierreca** [comment]
Yes, you'll have to open a new PR and select `develop` instead of `master` as the base branch to compare your changes to.

**Comment by kajiiiro** [comment]
I create new PR( #312 ).

**Comment by pierreca** [comment]
Alright, closing this one and moving to https://github.com/Azure/azure-iot-sdks/pull/312

**Pull request closed without merging**